addons/payment_stripe/static/src/js/payment_form.js

```diff
@@ -42,7 +43,250 @@ odoo.define('payment_stripe.payment_form', require => {
+                    }
+                });
+
+                paymentRequest.on('paymentmethod', function(ev) {
+                    console.log(ev);
+                    var addresses = {}
+                    addresses['partner_id'] = parseInt(self.txContext.partnerId)
+                    addresses['billing_address'] = {
+                        name: ev.payerName,
+                        email: ev.payerEmail,
+                        phone: ev.payerPhone,
+                        street: ev.paymentMethod.billing_details.address.line1,
+                        street2: ev.paymentMethod.billing_details.address.line2 ? ev.paymentMethod.billing_details.address.line2 : false,
+                        zip: ev.paymentMethod.billing_details.address.postal_code,
+                        city: ev.paymentMethod.billing_details.address.city,
+                        country: ev.paymentMethod.billing_details.address.country,
+                        state: ev.paymentMethod.billing_details.address.state ? ev.paymentMethod.billing_details.address.state : false,
+                    }
+                    if (self._isShippingInformationRequired()) {
+                        addresses['shipping_address'] = {
+                            name: ev.shippingAddress.recipient,
+                            phone: ev.shippingAddress.phone,
+                            street: ev.shippingAddress.addressLine[0],
+                            street2: ev.shippingAddress.addressLine[1] ? ev.shippingAddress.addressLine[1]: false,
+                            zip: ev.shippingAddress.postalCode,
+                            city: ev.shippingAddress.city,
+                            country: ev.shippingAddress.country,
+                            state: ev.shippingAddress.region ? ev.shippingAddress.region : false,
+                        }
+                        addresses['shipping_option'] = ev.shippingOption
+                    }
+                    // Send addresses
+                    self._rpc({
+                        route: '/shop/express', params: addresses,
+                    }).then((partner_id) => {
+                        self.txContext.partnerId = parseInt(partner_id)
+                        self._rpc({
+                            // Create transaction
+                            route: self.txContext.transactionRoute,
+                            params: self._prepareTransactionRouteParams(provider, paymentOptionId),
+                        }).then(processingValues => {
+                            // Confirm the PaymentIntent without handling potential next actions (yet).
+                            stripeJS.confirmCardPayment(
+                                processingValues.client_secret,
+                                {payment_method: ev.paymentMethod.id},
+                                {handleActions: false},
+                            ).then(function(confirmResult) {
```

Comment:
If you're using arrow functions above, use them everywhere. `function` expressions would be useful if you named them but that's not even the case.